### PR TITLE
Support for loading dictionaries

### PR DESCRIPTION
**Pull request by pawelknes**
Added loader which can be used to load dictionaries as per #699

### CODE REVIEW

File: src/unitxt/loaders.py [hunk 1]
```diff
@@ -474,3 +474,21 @@ def process(self):
         return FixedFusion(
             origins=self.sources, max_instances_per_origin=self.get_limit()
         ).process()
+
+
+class LoadDirectory(Loader):
```
Comment (yoavkatz) [inline src/unitxt/loaders.py:479]: I think it should be either LoadDictionary or LoadFromConstants.

Please also add a documentation string with example of how to use it, so will be appear in the auto generated documentation.

File: tests/library/test_loaders.py [hunk 2]
```diff
@@ -226,3 +232,20 @@ def test_multiple_source_loader(self):
...
+                {"input": "Input2", "output": "Result2"},
+            ],
+            "test": [
+                {"input": "Input3", "output": "Result3"},
+            ],
+        }
+        loader = LoadDirectory(data=data)
+        streams = loader.process()
+
+        for data_type, instances in data.items():
```
Comment (yoavkatz) [inline tests/library/test_loaders.py:249]:
```suggestion
        for split, instances in data.items():
```

File: tests/library/test_loaders.py [hunk 3]
```diff
@@ -226,3 +232,20 @@ def test_multiple_source_loader(self):
...
+            for instance, stream in zip(instances, streams[data_type]):
```
Comment (yoavkatz) [inline tests/library/test_loaders.py:250]:
```suggestion
            for original_instance, stream_instance in zip(instances, streams[data_type]):
```

**Review by yoavkatz — CHANGES_REQUESTED** [review]
Thanks. Very useful class. I added a few comments.

File: src/unitxt/loaders.py [hunk 4]
```diff
@@ -474,3 +474,37 @@ def process(self):
...
+    """
+
+    data: Dict[str, List[Dict[str, Any]]]
+
+    @staticmethod
+    def yield_instance(instances: List[Dict[str, Any]]) -> Dict[str, Any]:
+        yield from instances
+
+    def process(self) -> MultiStream:
+        return MultiStream(
```
Comment (dafnapension) [inline src/unitxt/loaders.py:503]: suggesting:
return MultiStream.from_iterables(self.data)

Comment (pawelknes) [reply src/unitxt/loaders.py]: Good point, thanks

File: tests/library/test_loaders.py [hunk 5]
```diff
@@ -226,3 +232,20 @@ def test_multiple_source_loader(self):
             )
             ms = loader()
             assert len(dfs["test"]) + len(dfs["train"]) == len(list(ms["test"]))
+
+    def test_load_from_directory(self):
```
Comment (yoavkatz) [inline tests/library/test_loaders.py:236]:
```suggestion
    def test_load_from_dictionary(self):
```

**Review by yoavkatz — APPROVED** [review]
Looks good thanks.

**Pull request merged**